public/dorequest.php

```diff
@@ -277,6 +337,78 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
         $response['AchievementID'] = $achIDToAward;
         break;
 
+    // This is only currently supported for "Standalone" integrations.
+    case "awardachievements":
+        if (request()->method() !== 'POST') {
+            return DoRequestError('Access denied.', 403, 'access_denied');
+        }
+
+        $achievementIdsInput = request()->input('a', '');
+        $validationHashesInput = request()->input('v', '');
+        $hardcore = (bool) request()->input('h', 0);
+
+        $foundTargetUser = null;
+        if (!$delegateTo) {
+            return DoRequestError("You must specify a target user.", 400);
+        }
+
+        $foundTargetUser = User::firstWhere('User', $delegateTo);
+        if (!$foundTargetUser) {
+            return DoRequestError("The target user couldn't be found.", 400);
+        }
+
+        $achievementIdsArray = explode(',', $achievementIdsInput);
+        $validationHashesArray = explode(',', $validationHashesInput);
```

Comment:
Recommend generating one validation hash for the CSV, rather than sending in a parallel CSV of validation hashes.

Reply:
That makes sense.

This function now accepts a single validation hash, constructed using all the given achievement IDs.